airbyte-webapp/src/components/base/Button/Button.tsx

```diff
@@ -1,134 +1,68 @@
-import styled from "styled-components";
-import { Theme } from "theme";
+import { faCircleNotch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import React from "react";
 
+import styles from "./Button.module.scss";
 import { ButtonProps } from "./types";
 
-type IStyleProps = ButtonProps & { theme: Theme };
-
-const getBorderColor = (props: IStyleProps) => {
-  if ((props.secondary && props.wasActive) || props.iconOnly) {
-    return "transparent";
-  }
-
-  if (props.secondary) {
-    return props.theme.greyColor30;
-  } else if (props.danger) {
-    return props.theme.dangerColor;
-  }
-
-  return props.theme.primaryColor;
-};
-
-const getBackgroundColor = (props: IStyleProps) => {
-  if (props.wasActive) {
-    if (props.secondary || props.iconOnly) {
-      return props.theme.primaryColor12;
-    }
-    return "transparent";
-  } else if (props.secondary || props.iconOnly) {
-    return "transparent";
-  } else if (props.danger) {
-    return props.theme.dangerColor;
-  }
-
-  return props.theme.primaryColor;
-};
-
-const getTextColor = (props: IStyleProps) => {
-  if (props.wasActive) {
-    if (props.danger) {
-      return props.theme.dangerColor;
-    }
-    return props.theme.primaryColor;
-  } else if (props.secondary || props.iconOnly) {
-    return props.theme.darkGreyColor;
-  }
-
-  return props.theme.whiteColor;
-};
-
-const getDisabledTextColor = (props: IStyleProps) => {
-  if (props.danger) {
-    return props.theme.dangerColor;
-  } else if (props.iconOnly) {
-    return props.theme.greyColor40;
+export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
+  const buttonStyles = {
+    [styles.full]: props.full,
+    [styles.isLoading]: props.isLoading,
+    [styles.sizeL]: props.size === "lg",
+    [styles.sizeS]: props.size === "sm",
+    [styles.sizeXS]: props.size === "xs",
+    [styles.typeDanger]: props.variant === "danger",
+    [styles.typeLight]: props.variant === "light",
+    [styles.typePrimary]: props.variant === "primary",
+    [styles.typeSecondary]: props.variant === "secondary",
+  };
+  const widthStyle: { width?: string } = {};
+  if (props.width) {
+    widthStyle.width = `${props.width}px`;
   }
-
-  return getTextColor(props);
-};
-
-const getDisabledOpacity = (props: IStyleProps) => {
-  if (props.danger) {
-    return ".5";
-  } else if (props.iconOnly) {
-    return "1";
-  }
-
-  return ".3";
+  return (
+    <button
+      ref={ref}
+      style={widthStyle}
+      className={classNames(styles.button, props.customStyles, buttonStyles)}
+      {...props}
```

Comment:
`iconPosition` sets as a attribute for native `<button>` html element

<img width="529" alt="image" src="https://user-images.githubusercontent.com/20929344/190208471-6bdbff28-f95a-4a79-9b25-a3a7c2ec505d.png">

Reply:
fixed